apps/dashboard/app/models/project.rb

```diff
@@ -194,15 +191,39 @@ def sync_template
     return true if template.blank?
 
     # Sync the template files over
-    oe, s = Open3.capture2e("rsync", "-a", "#{template}/", "#{project_dataroot}")
+    oe, s = Open3.capture2e(*rsync_args)
     raise oe unless s.success?
 
-    true
+    save_new_scripts
   rescue StandardError => e
     errors.add(:save, "Failed to sync template: #{e.message}")
     false
   end
 
+  # When copying a project from a template, we need new Script objects
+  # that point to the _new_ project directory, not the template's directory.
+  # This creates them _and_ serializes them to yml in the new directory.
+  def save_new_scripts
+    dir = Script.scripts_dir(template)
+    Dir.glob("#{dir}/*/form.yml").map do |script_yml|
+      Script.from_yaml(script_yml, project_dataroot)
+    end.map do |script|
+      saved_successfully = script.save
+      errors.add(:save, script.errors.full_messages) unless saved_successfully
+
+      saved_successfully
+    end.none? do |saved_successfully|
+      saved_successfully == false
```

Comment:
What is the equality test for? My thinking was since `script.save` => `Bool` that the idea would be to set `saved_successfully` to `false` since the array is empty, but I think I'm missing the idea.

Reply:
>the idea would be to set saved_successfully to false since the array is empty,

I don't follow this sentence, mostly because arrays aren't empty. `none?` isn't checking for `no values` (i.e., empty arrays) it's checking for `no values match this criteria`, i.e., no values are false.

But the first bit is `map`ping Script objects to booleans. So you'd get something like `[true, true, false]` if you tried to save 3 scripts and failed to save the last one. 

This expression then is looping over `[true, true, false]` and in English asks `Are there no false values?`- i.e., script saving failures.
```ruby
.none? do |saved_successfully|
      saved_successfully == false
end
```
it'll return `true` if there are no false values - no save errors, or false if there are at least 1 false value.

I guess maybe it would read better if it were reversed - calling `all?` and checking for true?
```ruby
.all? do |saved_successfully|
      saved_successfully == true
end
```

Reply:
The `all?` reads much better and makes instant sense. I hadn't used `none?` before and misunderstood its use, I thought you were checking if there were no entries :facepalm: